File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.js
/**
 * @param {number} k
 * @return {character}
 */
var kthCharacter = function(k) {
  let word = "a";
  while (word.length < k) {
    let temp = "";
    for (let ch of word) {
        let val = 97 + (ch.charCodeAt(0) - 97 + 1)%26;
        temp += String.fromCharCode(val);
    }
    word += temp;
  }
  return word[k-1];  
};

File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        word = "a"
        while len(word) < k:
            temp = ""
            for ch in word:
                # temp += chr(97 + ((ord(ch)-97)+1)%26)
                asc = ord(ch) + 1 if ch != "z" else ord(ch)
                temp += chr(asc)
            word += temp
     
        return word[k-1]